### ISSUE TITLE
Port the code to ESP32

### ISSUE THREAD

**Issue by ivanvach** [mKeRix/room-assistant#842]
**Is your feature request related to a problem? Please describe.**
Double your user base by porting the code to ESP32. 

**Describe the solution you'd like**
Same great functionality to be available in ESP32 microcontroller. ESP32 has WiFi , BT classic and BLE. It is cheaper and more widely available.  As the name of the product says it "room-assistant" it is needed at least one per room. The cost adds up and cheaper controller could be beneficial to users. 

**Describe alternatives you've considered**
Raspberry Pi Zero W is current solution. Restriction apply on purchases and at the end you pay 2-3 times more than ESP32.

**Additional context**

**Comment by CobraDunn** [comment]
Strongly 2nd this request.

**Comment by smmason** [comment]
Oh my, this would be awesome!

**Comment by DTTerastar** [comment]
https://espresense.com/ attempts to be this.  Let me know what I'm missing...

**Comment by corgan2222** [comment]
> 
> 
> https://espresense.com/ attempts to be this. Let me know what I'm missing...

OMG thanks a lot!
As is started with room presence detection, this esp project was not around and I'm so happy that you brought my attention to it!

For month room assistant is not working  correct, because of the known Bluetooth problems of all raspberry and the cluster goes offline if one node is missing. I would say, it work 1 out of 10 times.

So I gave espresense a try, and I'm more than happy! 

I use tree ESP32 to track my iPhone 10 and one Oneplus8 Android and have absolute zero problems! The detection range is outstanding, nearly 15 m around the corner from our house. And the best thing, it works every time with no downtime!

At first, it was a little tricky to find the right devices, but this has to be done only once.
Sadly the integration has no home-assistant auto discovery, so that I had to create my own sensors, but that's not a big deal. 

So thanks again for pointing me to that project!

**Comment by DTTerastar** [comment]
If you know any math/python I'd love help on my attempts at trilateration using ESPresense: https://github.com/ESPresense/ad-espresense-ips

**Comment by davewongillies** [comment]
> > https://espresense.com/ attempts to be this. Let me know what I'm missing...
> 
> Sadly the integration has no home-assistant auto discovery, so that I had to create my own sensors, but that's not a big deal.

Seems like they just merged auto discovery into espresense https://github.com/ESPresense/ESPresense/issues/16

**Comment by smmason** [comment]
> https://espresense.com/ attempts to be this. Let me know what I'm missing...

Holy crap, this looks awesome.  Thanks!
How did I miss this?

**Comment by CobraDunn** [comment]
@smmason - Don’t get too excited if you want to use an iPhone. It doesn’t work.

**Comment by Scott8586** [comment]
@corgan2222 - but it can't distinguish between two of the same or similar model iPhone at the same os level,  it's not a unique fingerprint. (based on observations of multiple idevices in my setup).  It doesn't yet take the place of classicBluetooth interrogation.

**Comment by corgan2222** [comment]
> 
> 
> @corgan2222 - but it can't distinguish between two of the same or similar model iPhone at the same os level, it's not a unique fingerprint. (based on observations of multiple idevices in my setup). It doesn't yet take the place of classicBluetooth interrogation.

As I only have one iPhone, I could not test this. God to know, thanks for the info.

**Comment by DTTerastar** [comment]
I just tested a new iphone, my ipad pro and for me everything get's different fingerprint.  In fact they seem to be ordinally counted based on the iCloud join time.  I don't know why it works for me and not everyone else...

**Comment by corgan2222** [comment]
@DTTerastar 

What's about the room-assistant companion app? Maybe we can use it? For me the app worked completely fine for detection. (I only had the problems on the receiver side with the crappy raspberry BT stack)

> room-assistant companion app for iOS

This app broadcasts BLE services for presence detection purposes. It allows room-assistant to detect your room presence even though the actually advertised address is randomized. 

- https://apps.apple.com/de/app/room-assistant/id1538642237
- https://github.com/mKeRix/room-assistant-companion-ios

**Comment by DTTerastar** [comment]
I can see those broadcasts, making them into a fingerprint now.

**Comment by corgan2222** [comment]
awesome!!!!

**Comment by ivanvach** [comment]
Just tried it using iPhone 12 as a beacon without any specific BT apps.
Noticed that "distance" varies when I move iPhone around but it is by far not accurate. 
Also I couldn't find any good documentation so I'm asking here
a) is there a good documentation of espresence ?
b) what are the meanings of the fields raw, speed, rssi@1m, rssi below
{"id":"apple:1006:10-7","rssi@1m":-59,"rssi":-75,"mac":"7ec32e9c6b6d","raw":2.87,"distance":3.13,"speed":-0.1}

I know that rssi is related to signal strength and I guess that distance is estimated buy rssi but it is so far off in my case. Is it possible to calibrate ?

**Comment by smmason** [comment]
> @smmason - Don’t get too excited if you want to use an iPhone. It doesn’t work.

Yeah, you're not wrong.... My phone and my wife's phone have the same fingerprint, as do our watches. Not giving up on it yet.

**Comment by corgan2222** [comment]
I can confirm the problem with my iWatch. It's the same fingerprint as my iPhone. :/

**Comment by smmason** [comment]
In fairness to the author of Room-assistant (which I still quite like by the way, its limitations are more Pi problems) should we really discuss this over on the  Espresense Github?

**Comment by DTTerastar** [comment]
https://github.com/ESPresense/ESPresense/releases/tag/v2.0.0 has added preliminary support for room assistant companion app!

**Comment by austwhite** [comment]
I've moved to using ESPresense, which is at 3.3.1 now and regularly updated.
Still would be nice if the Room Assistant guys were able to port this, but I have a feeling development on Room Assistant has very much slowed down.   This can happen as people get busy with other things.....
Thanks to the developers for letting us have it so long though :)
